source/packages/oos_util_string.pkb

```diff
@@ -331,7 +331,106 @@ as
     return l_return;
 
   end sprintf;
-
+  
+  
+  /**
+   * Does string replacement of keys by values
+   *
+   * @example
+   * SELECT oos_util_string.sprintf('Hello {firstName} {lastName}!', 
+   *                                 t_tab_key_value(t_rec_key_value('firstName', 'Nuno'),
+   *                                                 t_rec_key_value('lastName', 'Alves')
+   *                                                 )
+   *                               ) demo
+   * FROM dual;
+   *
+   * DEMO
+   * ------------------------------
+   * Hello Nuno Alves!
+   *
+   * @issue #134
+   *
+   * @author Nuno Alves
+   * @created 12-Nov-2016
+   * @param p_str Messsage to format using %s and %d replacement strings
+   * @param p_key_values Nested table of key value pairs to be replaced
+   * @param p_left_pattern Left string pattern, '{' by default
+   * @param p_right_pattern Right string pattern, '}' by default
+   * @return p_msg with strings replaced
+   */
+  function sprintf(
+    p_str in varchar2,
+    p_key_values in t_tab_key_value,
+    p_left_pattern in varchar2 default '{',
+    p_right_pattern in varchar2 default '}')
+    return varchar2
+  as
+    l_return varchar2(4000);
+  begin
+    l_return := p_str;
+    if p_key_values IS NOT NULL AND p_key_values.exists(1)
+    then
+      for i in 1 .. p_key_values.count
+      loop
+        l_return := REPLACE(l_return, p_left_pattern || p_key_values(i).key || p_right_pattern, p_key_values(i).value);
+      end loop;
+    end if;
+    
+    return l_return;
+  end sprintf;
+  
+  
+  /**
```

Comment:
Since all the code is in one PR I'll accept this but remove this function. It's already handled in standard sprintf